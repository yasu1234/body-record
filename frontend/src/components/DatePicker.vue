<script setup>
import { ref } from 'vue';
import VueDatePicker from '@vuepic/vue-datepicker';
import '@vuepic/vue-datepicker/dist/main.css';
import { format } from 'date-fns';

</script>

<template>
    <VueDatePicker
    v-model="date"
    locale="ja"
    week-start="0"
    format="yyyy/M/d"
    auto-apply
    :month-change-on-scroll="false"
    :enable-time-picker="false"
    @update:model-value="updateDate"/>
</template>

<style>
</style>