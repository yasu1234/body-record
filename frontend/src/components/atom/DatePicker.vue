<script setup>
import { ref } from 'vue';
import VueDatePicker from '@vuepic/vue-datepicker';
import '@vuepic/vue-datepicker/dist/main.css';
import { format } from 'date-fns';

const props = defineProps({
    date: String
})

const emit = defineEmits(['update:date']);

function updateDate(newDate) {
  emit('update:date', format(newDate, 'yyyy/M/d'));
}

</script>

<template>
    <VueDatePicker
    locale="ja"
    v-model="props.date"
    week-start="0"
    format="yyyy/M/d"
    auto-apply
    :month-change-on-scroll="false"
    :enable-time-picker="false"
    :date="date"
    @update:model-value="updateDate"/>
</template>

<style scoped>
</style>