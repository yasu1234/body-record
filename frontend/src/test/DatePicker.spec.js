import { shallowMount } from '@vue/test-utils'
import { useRouter } from "vue-router";
import DatePicker from '@/components/atom/DatePicker.vue'

jest.mock('vue-router', () => {
  const router = {
    push: jest.fn(),
    pathname: '/',
    route: '',
    asPath: '',
    query: {},
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
  };
  return {
    useRouter: jest.fn().mockReturnValue(router),
  };
});

describe('Header.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(DatePicker, {
      global: {
        stubs: ["router-link", "router-view"]
      }
    })
  })

  it('SnapShotテスト', () => {
    expect(wrapper.element).toMatchSnapshot();
  })

  it('emitsで指定した日付が渡る', async () => {
    const datePicker = wrapper.findComponent(DatePicker)
    await datePicker.vm.$emit('update:model-value', new Date(2023, 4, 15))
    expect(wrapper.emitted('update:date')).toBeTruthy()
    expect(wrapper.emitted('update:date')[0]).toEqual(['2023/5/15'])
  })

  it('emitsでnullを渡した時には空文字が渡る', async () => {
    const datePicker = wrapper.findComponent(DatePicker)
    await datePicker.vm.$emit('update:model-value', null)
    expect(wrapper.emitted('update:date')).toBeTruthy()
    expect(wrapper.emitted('update:date')[0]).toEqual([''])
  })

  it('propsが更新されたら更新した日づけに変更される', async () => {
    wrapper = shallowMount(DatePicker, {
      props: {
        date: '2023/4/1'
      }
    })
    expect(wrapper.props().date).toBe('2023/4/1')
    await wrapper.setProps({ date: '2023/5/15' })
    expect(wrapper.props().date).toBe('2023/5/15')
  })
})
