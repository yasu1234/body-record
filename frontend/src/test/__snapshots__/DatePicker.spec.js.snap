// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue SnapShotテスト 1`] = `
<vue-date-picker-stub
  actionrow="[object Object]"
  arialabels="[object Object]"
  arrownavigation="false"
  autoapply="true"
  autocomplete="off"
  autoposition="true"
  canceltext="Cancel"
  clearable="true"
  dark="false"
  disabled="false"
  disableddates=""
  disabledweekdays=""
  disablemonthyearselect="false"
  disabletimerangevalidation="false"
  disableyearselect="false"
  enableminutes="true"
  enableseconds="false"
  enabletimepicker="false"
  escclose="true"
  filters="[object Object]"
  fixedend="false"
  fixedstart="false"
  flow=""
  focusstartdate="false"
  format="yyyy/M/d"
  hideinputicon="false"
  hidenavigation=""
  hideoffsetdates="false"
  hoursgridincrement="1"
  hoursincrement="1"
  ignoretimevalidation="false"
  inline="false"
  is24="true"
  loading="false"
  locale="ja"
  markers=""
  minutesgridincrement="5"
  minutesincrement="1"
  modelauto="false"
  monthchangeonarrows="true"
  monthchangeonscroll="false"
  monthnameformat="short"
  monthpicker="false"
  multidates="false"
  nodisabledrange="false"
  nohoursoverlay="false"
  nominutesoverlay="false"
  nosecondsoverlay="false"
  notoday="false"
  nowbuttonlabel="Now"
  offset="10"
  partialflow="false"
  partialrange="true"
  placeholder=""
  position="center"
  presetdates=""
  preventminmaxnavigation="false"
  previewformat=""
  quarterpicker="false"
  range="false"
  required="false"
  reverseyears="false"
  secondsgridincrement="5"
  secondsincrement="1"
  selecttext="Select"
  showlastinrange="true"
  sixweeks="false"
  spaceconfirm="true"
  teleportcenter="false"
  textinput="false"
  timepicker="false"
  timepickerinline="false"
  transitions="true"
  ui="[object Object]"
  utc="false"
  vertical="false"
  weeknumname="W"
  weekpicker="false"
  weekstart="0"
  yearfirst="false"
  yearpicker="false"
  yearrange="1900,2100"
/>
`;
